src/engine/strat_engine/backstore/metadata/mda.rs

```diff
@@ -197,15 +204,21 @@ impl MDARegions {
 
         Ok(MDARegions {
             region_size,
-            mdas: [None, None],
+            mda_headers: [None, None],
         })
     }
 
-    /// Construct MDARegions from data on the disk.
-    /// Note that this method is always called in a context where a
-    /// StaticHeader has already been read. Therefore, it
-    /// constitutes an error if it is not possible to discover two
-    /// well-formed MDAHeaders for this device.
+    /// Construct an MDARegions struct from data on the disk.
+    /// The individual MDAHeaders in the struct may all be None, as it is
+    /// possible that no variable length metadata has been written to the
+    /// device on which the metadata has been written.
+    ///
+    /// Returns an error if there is an I/O error or if the MDA header data
+    /// on the device is invalid.
+    //
+    // TODO: Consider whether the return type of this method should be
```

Comment:
Create an issue so this isn't forgotten? (returning a `Result` containing an enum looking like this: `enum Error<E: Error, T> { IOError(E), Error(E), Ok(T) }` should do the trick I guess?)

Reply:
I thought of creating an issue, but I don't think that the question has quite that status yet. It's more a hint to the reader at this point, since I'm not sure if it would or would not be useful to distinguish in that way.

Reply:
Fine by me.